### ISSUE TITLE
Firestore runTransaction resolves to undefined

### ISSUE THREAD

**Issue by fshe** [soumak77/firebase-mock#86]
On success, `runTransaction` should return the promise returned by the update function. This way we can pass data out of the transaction:
https://firebase.google.com/docs/reference/js/firebase.firestore.Firestore#runTransaction

Example here:
https://firebase.google.com/docs/firestore/manage-data/transactions#passing_information_out_of_transactions

In the `then` callback of the promise returned by `runTransaction` the value isn't passed through when mocked, but works correctly when actually connected to Firestore. Perhaps I have done something wrong, so I've attached code for you to view as well:
```javascript
  db.runTransaction((transaction) => {
    const docRef = db.collection('items')
      .doc(itemId);

    return transaction.get(docRef).then((doc) => {
      if (doc.exists) {
        const currCount = doc.data().count;
        transaction.set(docRef, { count: currCount + 1 });
        return currCount + 1;
      }

      return 0;
    });
  }).then((newCount) => {
    console.log(newCount);
  });
```

Thank you in advance :)

**Comment by soumak77** [comment]
this will be fixed by #96

**Comment by soumak77** [comment]
fix available in [v2.2.6](https://github.com/soumak77/firebase-mock/releases/tag/v2.2.6)
